api/samples/cbrtrace.c

```diff
@@ -44,19 +44,24 @@
 #include "dr_api.h"
 #include "drmgr.h"
 #include "utils.h"
+#include "assert.h"
 
 static client_id_t client_id;
 
 static int tls_idx;
-
+static long cbrn = 0;
+static long cbrn_t = 0;
 /* Clean call for the cbr */
 static void
 at_cbr(app_pc inst_addr, app_pc targ_addr, app_pc fall_addr, int taken, void *bb_addr)
 {
+    cbrn += 1;
```

Comment:
This global variable increment is racy and will be inaccurate for many-threaded apps.  Best to use thread-local counters and aggregate at thread exit time under a lock (see e.g. instrace_simple num_refs).